### PR TITLE
Percentage Input (Vanilla)

### PR DESCRIPTION
**Pull request by bobwise**
## What does this PR do?
Adds the Vanilla Percentage Input to the Spark documentation site.

### Associated Issue 
Fixes https://github.com/sparkdesignsystem/spark-design-system/issues/1639
Fixes https://github.com/sparkdesignsystem/spark-design-system/issues/1771

### Documentation
 - [x] Update Spark Docs Vanilla
 - [x] Update Component Sass Var/Class Modifier table

### Code
 - [x] Build Component in Spark Vanilla (Sass, HTML, JS)
 - [x] Unit Testing in Spark Vanilla with `npm run test` (100% coverage, 100% passing) (no new tests)

### Accessibility
- [x] New changes abide by [accessibility requirements](https://sparkdesignsystem.com/docs/accessibility)

### Browser Testing (current version and 1 prior)
  - [x] Google Chrome
  - [x] Google Chrome (Mobile)
  - [x] Mozilla Firefox
  - [x] Mozilla Firefox (Mobile)
  - [x] Microsoft Internet Explorer 11 (only this specific version of IE)
  - [x] Microsoft Edge
  - [x] Apple Safari
  - [x] Apple Safari (Mobile)
  
### Manual Testing Plan
  - [x] Update excel sheet with component changes

### Design Review
 - [x] Design reviewed and approved

### Screenshots
image

### CODE REVIEW

File: packages/spark/base/inputs/_text-input.scss [hunk 1]
```diff
@@ -146,17 +162,24 @@ $sprk-input-huge-box-shadow: 0 3px 10px 1px rgba(0, 0, 0, 0.08) !default;
...
   opacity: $sprk-text-input-huge-placeholder-active-opacity;
 }
 
-.sprk-b-InputContainer--huge > :first-child.sprk-b-Label + .sprk-b-TextInputIconContainer .sprk-b-TextInput:focus,
-.sprk-b-InputContainer--huge .sprk-b-TextInputIconContainer .sprk-b-TextInput.sprk-b-TextInput--label-hidden:focus {
+.sprk-b-InputContainer--huge > :first-child.sprk-b-Label + .sprk-b-TextInputIconContainer
+.sprk-b-TextInput:focus,
+.sprk-b-InputContainer--huge
+.sprk-b-TextInputIconContainer
+.sprk-b-TextInput.sprk-b-TextInput--label-hidden:focus {
```
Comment (RVMendoza) [inline packages/spark/base/inputs/_text-input.scss]: Thanks for cleaning these up!

File: packages/spark/base/inputs/_text-icon-input.scss [hunk 2]
```diff
@@ -12,6 +12,22 @@
   }
 }
 
+.sprk-b-TextInputIconContainer--has-text-icon-right {
```
Comment (afebbraro) [inline packages/spark/base/inputs/_text-icon-input.scss]: Hey @bobwise , we could make this code a little more extendable by breaking it out like this. Let me know your thoughts on this!

```
.sprk-b-TextInputIconContainer--has-text-icon,
.sprk-b-TextInputIconContainer--has-text-icon-right {
  &::before {
    content: $sprk-monetary-symbol;
    display: block;
    font-size: $sprk-input-text-icon-font-size;
    font-weight: $sprk-input-text-icon-font-weight;
    line-height: 1;
    margin-left: $sprk-input-text-icon-offset-x;
    position: absolute;
    top: calc(100% - #{$sprk-input-text-icon-offset-y});
    z-index: $sprk-input-text-icon-z-index;
  }
}

.sprk-b-TextInputIconContainer--has-text-icon-right {
  &::before {
    content: $sprk-monetary-symbol;
  }
}

.sprk-b-TextInputIconContainer--narrow {
  max-width: $sprk-input-container-narrow-max-width;
}

```

This allows the use of the new `sprk-b-TextInputIconContainer--narrow` class to apply the max-width setting you have. Having this allows apps to apply the narrow max-width if they want to other icon inputs or to remove that class if they are building their own textinput with a special icon.

For the other part, chaining the two selectors `.sprk-b-TextInputIconContainer--has-text-icon` and 
`.sprk-b-TextInputIconContainer--has-text-icon-right` allows the removal of all the duplicated ::before code to setup the icon.

Comment (bobwise) [reply packages/spark/base/inputs/_text-icon-input.scss]: I added a new commit that consolidates all of the duplicated `::before` code. I think adding a new `sprk-b-TextInputIconContainer--narrow` class is scope creep. I'm going to be creating a new Issue to refactor this CSS to allow
- any input can be half-width, regardless of whether or not it has an icon
- inputs with an icon on the left can have any icon, not just $
- inputs with an icon on the right can have any icon, not just %

Comment (afebbraro) [reply packages/spark/base/inputs/_text-icon-input.scss]: @bobwise , thank you! Please link to the new issue here for future reference once you get that added.

File: packages/spark/base/inputs/_text-input.scss [hunk 3]
```diff
@@ -20,6 +20,162 @@
   border-color: $sprk-text-input-focus-border-color;
 }
 
+// Applies to any huge input
+.sprk-b-InputContainer--huge .sprk-b-TextInput {
+  position: relative;
```
Comment (guern1kn) [inline packages/spark/base/inputs/_text-input.scss]: We will want to make sure that the css for the huge inputs doesn't get put back into this file. Amber moved it all to the input_container file in a recent merge.

File: src/patterns/components/inputs/percentage-error.hbs [hunk 4]
```diff
@@ -0,0 +1,15 @@
+---
+name: percentage Input - Error State
```
Comment (guern1kn) [inline src/patterns/components/inputs/percentage-error.hbs]: Capitalize Percentage?

**Comment by bobwise** [comment]
Issue to make the CSS a little more flexible:

https://github.com/sparkdesignsystem/spark-design-system/issues/1784

File: src/patterns/components/inputs/date-disabled.hbs [hunk 5]
```diff
@@ -6,7 +6,7 @@ information:
   - Spark recommends that this component be used when you expect the entered date to be in the past.
   - Masking is automatically applied when the field contains a valid input.
   - The field value contains special characters which you may need to parse out before submitting the form.
-order: 28
+order: 30
```
Comment (afebbraro) [inline src/patterns/components/inputs/date-disabled.hbs]: I would double check that with my reordering that this is the right order. I don't remember 100% but I thought they only incremented by 1 and I see 2 here. I'm not sure.

File: src/patterns/components/inputs/percentage.hbs [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+name: Percentage Input
+description:
+  This input is used to collect a percent value.
+information:
+  - Spark does not apply any validation or formatting to these fields.
+order: 23
+---
+
+<div class="sprk-b-InputContainer" data-sprk-input="percentage">
+  <div class="sprk-b-TextInputIconContainer sprk-b-TextInputIconContainer--has-text-icon-right">
```
Comment (afebbraro) [inline src/patterns/components/inputs/percentage.hbs]: Hey @bobwise, this icon isn't a text icon anymore. Its a SVG so the class name of sprk-b-TextInputIconContainer--has-text-icon-right doesn't work. Needs some updating there.

File: src/patterns/components/inputs/percentage.hbs [hunk 7]
```diff
@@ -0,0 +1,34 @@
...
+    <label for="{{#if idattr}}{{idattr}}{{else}}percentage-normal{{/if}}"
+      class="sprk-b-Label sprk-b-Label--percentage">Interest Rate</label>
```
Comment (afebbraro) [inline src/patterns/components/inputs/percentage.hbs]: I don't see this class in the Sass anywhere, sprk-b-Label--percentage. What's it applying?

File: packages/spark/base/inputs/_text-icon-input.scss [hunk 8]
```diff
@@ -1,6 +1,6 @@
-.sprk-b-TextInputIconContainer--has-text-icon {
+.sprk-b-TextInputIconContainer--has-text-icon,
+.sprk-b-TextInputIconContainer--has-text-icon-right {
```
Comment (RCopeland) [inline packages/spark/base/inputs/_text-icon-input.scss]: Does this class need to be added here? We switched to use an svg instead, so it shouldnt need any of the styles on the ::before element, right?

File: packages/spark/base/inputs/_text-input.scss [hunk 9]
```diff
@@ -1,7 +1,9 @@
 // ==================================================================
 // Input Component
 // ==================================================================
-.sprk-b-TextInput {
+.sprk-b-TextInput,
+// sprk-b-TextInput to be deprecated and replace with sprk-b-InputContainer__input
```
Comment (afebbraro) [inline packages/spark/base/inputs/_text-input.scss]: could add a TODO here in front of the comment

File: packages/spark/settings/_settings.scss [hunk 10]
```diff
@@ -431,6 +431,8 @@ $sprk-list-indented-margin-left: $sprk-space-l !default;
 $sprk-monetary-symbol: '$' !default;
 $sprk-fieldset-padding: $sprk-space-m 0 0 0 !default;
 $sprk-input-container-margin: 0 0 $sprk-space-l 0 !default;
+$sprk-input-container-huge-max-width: 37.5rem !default;
```
Comment (afebbraro) [inline packages/spark/settings/_settings.scss]: This needs to be removed, might of been an artifact of the rebase/pull issue.

**Pull request merged**